apps/web/app/scripts/fetch-posts.js

```diff
@@ -0,0 +1,97 @@
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+/**
+ * Fetches all posts from the CMS by iterating through all pages
+ */
+async function fetchAllPosts() {
+  const allPosts = [];
+  let currentPage = 1;
+  let hasNextPage = true;
+  
+  console.log('Fetching posts from CMS...');
+  
+  while (hasNextPage) {
+    try {
+      console.log(`Fetching page ${currentPage}...`);
+      const response = await fetch(`https://cms.leather.io/posts?page=${currentPage}`);
+      
+      if (!response.ok) {
+        throw new Error(`Failed to fetch posts: ${response.status} ${response.statusText}`);
+      }
+      
+      const data = await response.json();
+      allPosts.push(...data.posts);
+      
+      hasNextPage = data.pagination.hasNextPage;
+      currentPage++;
+      
+      console.log(`Fetched ${data.posts.length} posts from page ${currentPage - 1}`);
+    } catch (error) {
+      console.error('Error fetching posts:', error);
+      hasNextPage = false;
+    }
+  }
+  
+  console.log(`Fetched a total of ${allPosts.length} posts`);
+  return allPosts;
+}
+
+/**
+ * Processes posts and converts them to a format indexed by slug
+ */
+function processPostsForStorage(posts) {
+  const postsMap = {};
+  
+  posts.forEach(post => {
+    // Skip posts without a slug
+    if (!post.Slug) {
```

Comment:
Something that should be enforced at the API layer I think.

Reply:
This is resolved in latest commit